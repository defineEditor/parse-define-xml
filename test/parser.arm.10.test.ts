import fs from "fs";
import path from "path";
import parseDefineXml from "parser/define.20";
import { MetaDataVersion20 } from "interfaces/arm.10";

describe("ARM 1.0 for Define-XML 2.0 Parser", () => {
    let xmlStringAdam: string;

    beforeAll(() => {
        // Load the sample Define-XML file
        xmlStringAdam = fs.readFileSync(path.join(__dirname, "data/define.adam.20.arm.xml"), "utf-8");
    });

    it("should have analysisResultDisplays", async () => {
        const defineXml = await parseDefineXml(xmlStringAdam, true);
        expect(defineXml.odm).toBeDefined();
        expect(defineXml.odm.study).toBeDefined();
        expect(defineXml.odm.study.studyOid).toBeTruthy();
        expect(defineXml.odm.study.globalVariables).toBeDefined();
        expect(defineXml.odm.study.metaDataVersion).toBeDefined();
        expect(defineXml.odm.study.metaDataVersion.analysisResultDisplays).toBeDefined();
    });

    describe("ARM Snapshots", () => {
        let mdv: MetaDataVersion20;
        beforeAll(async () => {
            const defineXml = await parseDefineXml(xmlStringAdam, true);
            mdv = defineXml.odm.study.metaDataVersion;
        });

        it("should match result display snapshot", () => {
            const rd = mdv.analysisResultDisplays.resultDisplays["RD.Table_14-5.02"];
            expect(rd).toMatchSnapshot();
        });
    });
});
