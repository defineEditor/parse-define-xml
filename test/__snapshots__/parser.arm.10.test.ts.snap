// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ARM 1.0 for Define-XML 2.0 Parser ARM Snapshots should match result display snapshot 1`] = `
{
  "analysisResultOrder": [
    "AR.Table_14-5.02.R.1",
  ],
  "analysisResults": {
    "AR.Table_14-5.02.R.1": {
      "analysisDatasets": {
        "analysisDatasets": {
          "IG.ADAE": {
            "analysisVariables": [
              "IT.ADAE.AEBODSYS",
              "IT.ADAE.AEDECOD",
            ],
            "itemGroupOid": "IG.ADAE",
            "whereClauseRefs": [
              "WC.Table_14-5.02.R.1.ADAE",
            ],
          },
          "IG.ADSL": {
            "itemGroupOid": "IG.ADSL",
            "whereClauseRefs": [
              "WC.Table_14-5.02.R.1.ADSL",
            ],
          },
        },
        "analysisDatasetsOrder": [
          "IG.ADAE",
          "IG.ADSL",
        ],
        "commentOid": "COM.JOIN-ADSL-ADAE",
      },
      "analysisPurpose": "PRIMARY OUTCOME MEASURE",
      "analysisReason": "SPECIFIED IN SAP",
      "description": [
        {
          "value": "Incidence of Treatment Emergent Serious Adverse Events by Treatment Group",
          "xml_lang": "en",
        },
      ],
      "documentation": {
        "description": [
          {
            "value": "Unique count of subjects that experienced an Adverse Event by Preferred Term, System Organ Class, and Treatment Group and percentages based on the number of subjects in the safety population within each treatment group. The total number of times an event occurred was recorded by Preferred Term, System Organ Class, and Treatment Group. Fisherâ€™s exact test was used for treatment comparison of event rates.
                ",
            "xml_lang": "en",
          },
        ],
        "documents": [
          {
            "leafId": "LF.SAP-SEC-11.2",
            "pdfPageRefs": [
              {
                "pageRefs": "5",
                "type": "PhysicalRef",
              },
            ],
          },
        ],
      },
      "oid": "AR.Table_14-5.02.R.1",
      "programmingCode": {},
    },
  },
  "description": [
    {
      "value": "Incidence of Treatment Emergent Serious Adverse Events by Treatment Group",
      "xml_lang": "en",
    },
  ],
  "documents": [
    {
      "leafId": "LF.Table-14-5.02",
      "pdfPageRefs": [
        {
          "pageRefs": "3",
          "type": "PhysicalRef",
        },
      ],
    },
  ],
  "name": "Table 14-5.02",
  "oid": "RD.Table_14-5.02",
}
`;
